sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PartialMethodNoImplementation.cs

```diff
@@ -74,29 +76,32 @@ private static void CheckForCandidatePartialDeclaration(SyntaxNodeAnalysisContex
         private static void CheckForCandidatePartialInvocation(SyntaxNodeAnalysisContext context)
         {
             var invocation = (InvocationExpressionSyntax)context.Node;
+
+            if (!(invocation.Parent is StatementSyntax statement))
+            {
+                return;
+            }
+
             if (!(context.SemanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol))
             {
                 return;
             }
 
             // from the method symbol it's not possible to tell if it's a partial method or not.
             // https://github.com/dotnet/roslyn/issues/48
-
             var partialDeclarations = methodSymbol.DeclaringSyntaxReferences
                 .Select(r => r.GetSyntax())
                 .OfType<MethodDeclarationSyntax>()
-                .Where(method => method.Body == null && method.Modifiers.Any(SyntaxKind.PartialKeyword));
+                .Where(method => method.Modifiers.Any(SyntaxKind.PartialKeyword) &&
+                    method.Body == null &&
+                    method.ExpressionBody == null);

```

Comment:
It was working without the new check but I think that's better to do early return whenever possible.